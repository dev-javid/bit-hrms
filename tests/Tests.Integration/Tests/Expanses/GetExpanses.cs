namespace Tests.Integration.Tests.Expanses;

public class GetExpanses(TestWebApplicationFactory<Program> factory) : IntegrationTest(factory)
{
    private const string Route = "api/expanses?page=1&limit=10";

    [Fact]
    [UseReporter(typeof(DiffReporter))]
    public async Task Should_return_expanses()
    {
        await LoginAsCompanyAdminAsync();
        await FeedDataAsync("Tests/Expanses/GetExpanses.sql");

        HttpResponseMessage response = await Client.GetAsync(Route);

        response.StatusCode.Should().Be(HttpStatusCode.OK);
        var result = await response.Content.ReadAsStringAsync();
        Approvals.VerifyJson(result);
    }
}
