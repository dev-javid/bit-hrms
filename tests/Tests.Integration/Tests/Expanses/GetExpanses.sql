INSERT INTO public.expanses (id, company_id, amount, documents, purpose, created_on, created_by, modified_by, modified_on)
VALUES 
    (1, 999, decode('r9a0jnes21bjwubV9WWhyD471a+GfkfaSxfWvm/EipI=', 'base64'), ARRAY['test.png', 'test2.png'], null, '2024-05-03 03:51:44.770234-07', 100, null, null),
    (2, 999, decode('NCreznXjp8Mj+MhlL7E1VJCTGtJ6yHRnlMW3SslYM74=', 'base64'), ARRAY['test.png', 'test2.png'], null, '2024-05-04 03:51:44.770234-07', 100, null, null),
    (3, 999, decode('r9a0jnes21bjwubV9WWhyD471a+GfkfaSxfWvm/EipI=', 'base64'), ARRAY['test.png', 'test2.png'], null, '2024-05-05 03:51:44.770234-07', 100, null, null),
    (4, 999, decode('NCreznXjp8Mj+MhlL7E1VJCTGtJ6yHRnlMW3SslYM74=', 'base64'), ARRAY['test.png', 'test2.png'], null, '2024-05-06 03:51:44.770234-07', 100, null, null),
    (5, 999, decode('r9a0jnes21bjwubV9WWhyD471a+GfkfaSxfWvm/EipI=', 'base64'), ARRAY['test.png', 'test2.png'], null, '2024-05-07 03:51:44.770234-07', 100, null, null),
    (6, 999, decode('NCreznXjp8Mj+MhlL7E1VJCTGtJ6yHRnlMW3SslYM74=', 'base64'), ARRAY['test.png', 'test2.png'], null, '2024-05-08 03:51:44.770234-07', 100, null, null),
    (7, 999, decode('r9a0jnes21bjwubV9WWhyD471a+GfkfaSxfWvm/EipI=', 'base64'), ARRAY['test.png', 'test2.png'], null, '2024-05-09 03:51:44.770234-07', 100, null, null),
    (8, 999, decode('NCreznXjp8Mj+MhlL7E1VJCTGtJ6yHRnlMW3SslYM74=', 'base64'), ARRAY['test.png', 'test2.png'], null, '2024-05-10 03:51:44.770234-07', 100, null, null),
    (9, 999, decode('r9a0jnes21bjwubV9WWhyD471a+GfkfaSxfWvm/EipI=', 'base64'), ARRAY['test.png', 'test2.png'], null, '2024-05-11 03:51:44.770234-07', 100, null, null),
    (10, 998, decode('NCreznXjp8Mj+MhlL7E1VJCTGtJ6yHRnlMW3SslYM74=', 'base64'), ARRAY['test.png', 'test2.png'], null, '2024-05-12 03:51:44.770234-07', 100, null, null),
    (11, 998, decode('r9a0jnes21bjwubV9WWhyD471a+GfkfaSxfWvm/EipI=', 'base64'), ARRAY['test.png', 'test2.png'], null, '2024-05-13 03:51:44.770234-07', 100, null, null),
    (12, 998, decode('NCreznXjp8Mj+MhlL7E1VJCTGtJ6yHRnlMW3SslYM74=', 'base64'), ARRAY['test.png', 'test2.png'], null, '2024-05-14 03:51:44.770234-07', 100, null, null),
    (13, 998, decode('r9a0jnes21bjwubV9WWhyD471a+GfkfaSxfWvm/EipI=', 'base64'), ARRAY['test.png', 'test2.png'], null, '2024-05-15 03:51:44.770234-07', 100, null, null),
    (14, 998, decode('NCreznXjp8Mj+MhlL7E1VJCTGtJ6yHRnlMW3SslYM74=', 'base64'), ARRAY['test.png', 'test2.png'], null, '2024-05-16 03:51:44.770234-07', 100, null, null),
    (15, 998, decode('r9a0jnes21bjwubV9WWhyD471a+GfkfaSxfWvm/EipI=', 'base64'), ARRAY['test.png', 'test2.png'], null, '2024-05-17 03:51:44.770234-07', 100, null, null),
    (16, 998, decode('NCreznXjp8Mj+MhlL7E1VJCTGtJ6yHRnlMW3SslYM74=', 'base64'), ARRAY['test.png', 'test2.png'], null, '2024-05-18 03:51:44.770234-07', 100, null, null);