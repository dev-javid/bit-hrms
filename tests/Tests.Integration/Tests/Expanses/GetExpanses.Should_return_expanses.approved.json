{
    "page": 1,
    "limit": 10,
    "total": 9,
    "items": [
        {
            "expanseId": 9,
            "companyId": 999,
            "amount": 1000,
            "documents": [
                "http://localhost/media/Expanses/9/test.png",
                "http://localhost/media/Expanses/9/test2.png"
            ],
            "purpose": null,
            "createdOn": "2024-05-11T10:51:44.770234Z"
        },
        {
            "expanseId": 8,
            "companyId": 999,
            "amount": 1200,
            "documents": [
                "http://localhost/media/Expanses/8/test.png",
                "http://localhost/media/Expanses/8/test2.png"
            ],
            "purpose": null,
            "createdOn": "2024-05-10T10:51:44.770234Z"
        },
        {
            "expanseId": 7,
            "companyId": 999,
            "amount": 1000,
            "documents": [
                "http://localhost/media/Expanses/7/test.png",
                "http://localhost/media/Expanses/7/test2.png"
            ],
            "purpose": null,
            "createdOn": "2024-05-09T10:51:44.770234Z"
        },
        {
            "expanseId": 6,
            "companyId": 999,
            "amount": 1200,
            "documents": [
                "http://localhost/media/Expanses/6/test.png",
                "http://localhost/media/Expanses/6/test2.png"
            ],
            "purpose": null,
            "createdOn": "2024-05-08T10:51:44.770234Z"
        },
        {
            "expanseId": 5,
            "companyId": 999,
            "amount": 1000,
            "documents": [
                "http://localhost/media/Expanses/5/test.png",
                "http://localhost/media/Expanses/5/test2.png"
            ],
            "purpose": null,
            "createdOn": "2024-05-07T10:51:44.770234Z"
        },
        {
            "expanseId": 4,
            "companyId": 999,
            "amount": 1200,
            "documents": [
                "http://localhost/media/Expanses/4/test.png",
                "http://localhost/media/Expanses/4/test2.png"
            ],
            "purpose": null,
            "createdOn": "2024-05-06T10:51:44.770234Z"
        },
        {
            "expanseId": 3,
            "companyId": 999,
            "amount": 1000,
            "documents": [
                "http://localhost/media/Expanses/3/test.png",
                "http://localhost/media/Expanses/3/test2.png"
            ],
            "purpose": null,
            "createdOn": "2024-05-05T10:51:44.770234Z"
        },
        {
            "expanseId": 2,
            "companyId": 999,
            "amount": 1200,
            "documents": [
                "http://localhost/media/Expanses/2/test.png",
                "http://localhost/media/Expanses/2/test2.png"
            ],
            "purpose": null,
            "createdOn": "2024-05-04T10:51:44.770234Z"
        },
        {
            "expanseId": 1,
            "companyId": 999,
            "amount": 1000,
            "documents": [
                "http://localhost/media/Expanses/1/test.png",
                "http://localhost/media/Expanses/1/test2.png"
            ],
            "purpose": null,
            "createdOn": "2024-05-03T10:51:44.770234Z"
        }
    ]
}